src/odemis/gui/cont/acquisition.py

```diff
@@ -1573,6 +1575,13 @@ def on_acquisition_done(self, future):
             self._reset_acquisition_gui("Acquisition failed (see log panel).", level=logging.WARNING)
             return
 
+    def _on_storage_dir(self, sdir):
+        sdir = sdir.split('@')[-1]  # remove username/password
+        sdir = sdir.split('/', maxsplit=1)[-1]  # remove host
+        sdir = "%s/%s" % (sdir, self.path)
+        logging.debug("New storage directory for image data: %s.", sdir)
+        self._tab_panel.txt_destination.SetValue(sdir)
```

Comment:
For now, externalStorageURL is readonly, so this shouldn't be necessary. If useful to display the whole path on the FTP (I'm personally not convinced), you could listen to only asm.filename (which contains the subfolders and the file name).

Reply:
Yes, I agree, this function can now go as we do not need to listen to the external storage anymore and the implementation as described above will be a lot simpler.